Diffusion/Common/DiffusionImage.swift

```diff
@@ -0,0 +1,132 @@
+//
+//  DiffusionImage.swift
+//  Diffusion
+//
+//  Created by Dolmere on 03/07/2023.
+//
+
+import SwiftUI
+import StableDiffusion
+import CoreTransferable
+
+/// Tracking for a `DiffusionImage` generation state.
+enum DiffusionImageState {
+    case generating
+    case waiting
+    case complete
+}
+
+/// Generic custom error to use when an image generation fails.
+enum DiffusionImageError: Error {
+    case invalidDiffusionImage
+}
+
+/// Combination of a `DiffusionImage` and its associated `DiffusionImageState`
+struct DiffusionImageWrapper {
+    var diffusionImageState: DiffusionImageState = .waiting
+    var diffusionImage: DiffusionImage? = nil
+}
+
+/// Model class to hold a generated image and the "recipe" data that was used to generate it
+final class DiffusionImage: NSObject, Identifiable, NSCoding, NSSecureCoding {
+    
+    let id: UUID
+    let cgImage: CGImage
+    let seed: UInt32
+    let steps: Double
+    let positivePrompt: String
+    let negativePrompt: String
+    let guidanceScale: Double
+    let disableSafety: Bool
+    /// Note: We created a local enum StableDiffusionScheduler with a String type so that we can track the Scheduler used in the image recipe. The StableDiffusion.StableDiffusionScheduler enum has not got a raw type and therefore cannot conform to NSSecureCoding which we use for Copy/Drag operations.
```

Comment:
```suggestion
    /// Local enum represented with a String to conform to NSSecureCoding
```

I think this might be a bit redundant given the comment in the type definition. Readers of this class may just need a pointer instead of all details.